core/src/main/java/org/mule/routing/AbstractSplitter.java

```diff
@@ -33,6 +32,6 @@
         return new CollectionMessageSequence(splitMessage(event));
     }
 
-    protected abstract List<MuleMessage> splitMessage(MuleEvent event) throws MuleException;
+    protected abstract List<MuleEvent> splitMessage(MuleEvent event) throws MuleException;
```

Comment:
The purpose of this method is still to split the message, as before, it just happens to receive event as context.